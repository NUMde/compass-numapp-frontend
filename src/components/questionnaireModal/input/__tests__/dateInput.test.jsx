import React from 'react';
import { Platform } from 'react-native';
import {
  renderWithRedux,
  act,
  fireEvent,
} from '../../../../../__test-utils__/render';

import DateInput from '../dateInput';

const initialState = {
  Questionnaire: {
    itemMap: {
      1: {
        linkId: '1',
        type: 'group',
        text: 'group',
        item: [{ linkId: '1.1', type: 'date' }],
      },
      1.1: { linkId: '1.1', type: 'date', text: 'date' },
    },
  },
};

describe('DateInput (Android/default)', () => {
  beforeAll(() => {
    Platform.OS = 'android';
  });

  it('should render a dateInput element', () => {
    const { toJSON } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should show datepicker when clicking input', () => {
    const { getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    expect(getByTestId('chosenDate')).toBeTruthy();
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    expect(getByTestId('DatePicker')).toBeTruthy();
  });

  it('should update value when date changes', () => {
    const { getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    act(() => {
      getByTestId('DatePicker').props.onChange({
        nativeEvent: { timestamp: new Date('Mon Mar 14 2022') },
      });
    });
    expect(getByTestId('chosenDate').props.value).toBe('14.03.2022');
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    act(() => {
      getByTestId('DatePicker').props.onChange({
        nativeEvent: {},
      });
    });
    expect(getByTestId('chosenDate').props.value).toBe('14.03.2022');
  });
});

describe('DateInput (iOS)', () => {
  beforeAll(() => {
    Platform.OS = 'ios';
  });

  it('should render a dateInput element', () => {
    const { toJSON } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should show datepicker when clicking input', () => {
    const { getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    expect(getByTestId('overlay')).toBeTruthy();
    act(() => {
      fireEvent.press(getByTestId('overlay'));
    });
    expect(getByTestId('DatePicker')).toBeTruthy();
  });

  it('should update value when date changes', () => {
    const { getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    act(() => {
      getByTestId('DatePicker').props.onChange({
        nativeEvent: { timestamp: new Date('Mon Mar 12 2022') },
      });
    });
    expect(getByTestId('chosenDate').props.value).toBe('12.03.2022');
  });

  it("should hide datepicker when clicking 'ok' button ", () => {
    const { queryByTestId, getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    expect(getByTestId('DatePicker')).toBeTruthy();
    act(() => {
      fireEvent.press(getByTestId('ios.submit'));
    });
    expect(queryByTestId('DatePicker')).toBeFalsy();
  });

  it("should hide datepicker and reset when clicking 'abort' button ", () => {
    const { queryByTestId, getByTestId } = renderWithRedux(
      <DateInput item={{ linkId: '1.1', type: 'date', text: 'date' }} />,
      { initialState },
    );
    act(() => {
      fireEvent.press(getByTestId('chosenDate'));
    });
    expect(getByTestId('DatePicker')).toBeTruthy();
    act(() => {
      fireEvent.press(getByTestId('ios.abort'));
    });
    expect(queryByTestId('DatePicker')).toBeFalsy();
    expect(getByTestId('chosenDate').props.value).toBe(null);
  });
});
