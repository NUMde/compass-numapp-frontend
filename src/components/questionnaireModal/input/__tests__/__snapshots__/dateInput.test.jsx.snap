// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput (Android/default) should render a dateInput element 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 33.75,
        "lineHeight": 43.125,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
  >
    date
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="overlay"
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#242424",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          editable={false}
          placeholderTextColor="#86939e"
          pointerEvents="none"
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="chosenDate"
          underlineColorAndroid="transparent"
          value={null}
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`DateInput (iOS) should render a dateInput element 1`] = `
<View
  style={
    Object {
      "marginBottom": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1F2532",
        "fontFamily": "IBMPlexSans-SemiBold",
        "fontSize": 33.75,
        "lineHeight": 43.125,
        "marginBottom": 5,
        "marginTop": 5,
      }
    }
  >
    date
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="overlay"
  >
    <View
      style={
        Object {
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <View
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "borderColor": "#86939e",
            "flexDirection": "row",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 40,
              "justifyContent": "center",
              "marginVertical": 4,
              "paddingRight": 4,
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#242424",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          editable={false}
          placeholderTextColor="#86939e"
          pointerEvents="none"
          style={
            Object {
              "color": "#242424",
              "flex": 1,
              "fontSize": 18,
              "minHeight": 40,
            }
          }
          testID="chosenDate"
          underlineColorAndroid="transparent"
          value={null}
        />
      </View>
      <Text
        style={
          Object {
            "color": "#ff190c",
            "fontSize": 12,
            "margin": 5,
          }
        }
      />
    </View>
  </View>
</View>
`;
